tests/manual-test-cases/Group5-Functional-Tests/5-10-Multiple-Datacenter.robot

```diff
@@ -17,11 +17,40 @@ Documentation  Test 5-10 - Multiple Datacenters
 Resource  ../../resources/Util.robot
 Suite Teardown  Run Keyword And Ignore Error  Nimbus Cleanup  ${list}
 
+*** Keywords ***
+# Insert elements from dict2 into dict1, overwriting conflicts in dict1 & returning new dict
+Combine Dictionaries
+    [Arguments]  ${dict1}  ${dict2}
+    ${dict2keys}=  Get Dictionary Keys  ${dict2}
+    :FOR  ${key}  IN  @{dict2keys}
+    \    ${elem}=  Get From Dictionary  ${dict2}  ${key}
+    \    Set To Dictionary  ${dict1}  ${key}  ${elem}
+    [Return]  ${dict1}
+
 *** Test Cases ***
 Test
     Log To Console  \nStarting test...
-    ${esx1}  ${esx1-ip}=  Deploy Nimbus ESXi Server  %{NIMBUS_USER}  %{NIMBUS_PASSWORD}
-    ${esx2}  ${esx2-ip}=  Deploy Nimbus ESXi Server  %{NIMBUS_USER}  %{NIMBUS_PASSWORD}
+    &{esxes}=  Create Dictionary
+    ${num_of_esxes}=  Evaluate  2
+    :FOR  ${i}  IN RANGE  3
+    # Deploy some ESXi instances
+    \    &{new_esxes}=  Deploy Multiple Nimbus ESXi Servers in Parallel  ${num_of_esxes}  %{NIMBUS_USER}  %{NIMBUS_PASSWORD}
+    \    ${esxes}=  Combine Dictionaries  ${esxes}  ${new_esxes}
+
+    # Investigate to see how many were actually deployed
+    \    ${len}=  Get Length  ${esxes}
+    \    ${num_of_esxes}=  Evaluate  ${num_of_esxes} - ${len}
+
+    # Exit if we've got enough & continue loop if we don't
+    \    Exit For Loop If  ${len} >= ${num_of_esxes}
```

Comment:
@mhagen-vmware, in the case where, during the first pass of this for loop, only one esx is deployed during the "Deploy Multiple Nimbus ESXi Servers in Parallel" (line 37), we could hit a condition where len = 1 and num_of_esxes = 1 causing the for loop to exit and then causing line 51 where 'esx2 = Get From List  ${esx-names}  1' could/will fail since we expected at least 2 esxes and not 1.
Can we get rid of the '=' here?

Also, on every new iteration of this for loop, does (num_of_esxes) get reset to 2?  <-- I'm just curious about that.  How does Robot handle (num_of_esxes) if we assign something to it inside the for loop and then we loop again?